src/codecs/tga/decoder.rs

```diff
@@ -238,6 +238,7 @@ impl<R: Read + Seek> TgaDecoder<R> {
     /// Reads a run length encoded data for given number of bytes
     fn read_encoded_data(&mut self, num_bytes: usize) -> io::Result<Vec<u8>> {
         let mut pixel_data = Vec::with_capacity(num_bytes);
+        let mut pixel = Vec::with_capacity(self.bytes_per_pixel);
```

Comment:
Can you rename this to `repeat_buf` to clarify its use and avoid confusion with `pixel_data`.